File: cs50/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void build_pyramids(int _brks);

int main(void)
{
    int bricks;
    do
    {
        bricks = get_int("Heigth:");
    }
    while (bricks <= 0);

    build_pyramids(bricks + 1);
}

void build_pyramids(int _brks)
{
    for (int i = 1; i < _brks; i++)
    {
        for (int j = 1; j < _brks - i; j++)
        {
            printf(" ");
        }
        for (int k = 0; k < i; k++)
        {
            printf("#");
        }
        printf("  ");

        for (int w = 0; w < i; w++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: cs50/sentimental-mario-more/mario.py

def get_int(prompt):

    while True:
        try:
            value = int(input(prompt))
            if value == 9:
                print("Invalid input. 9 is not allowed.")
            else:
                return value
        except ValueError:
            print("Invalid input. Please enter an integer.")

def build_pyramids(bricks):
    for i in range(1, bricks):
        for _ in range(bricks - i - 1):
            print(" ", end="")
        for _ in range(i):
            print("#", end="")
        print("  ", end="")
        for _ in range(i):
            print("#", end="")
        print()

def main():

    bricks = 0
    while bricks <= 0:
        bricks = get_int("Height: ")

    build_pyramids(bricks + 1)


main() # Acces Gate



